lib/matplotlib/animation.py

```diff
@@ -1545,6 +1546,12 @@ def func(frame, *fargs) -> iterable_of_artists
         - If an iterable, then simply use the values provided.  If the
           iterable has a length, it will override the *save_count* kwarg.
 
+          Note that when using ``repeat=True`` (the default) the iterable must
+          be usable multiple times (either ``iter(iterable) is not iterable``
```

Comment:
This is *really* inscrutable and very jargon-y.  Can you provide examples of what works and doesn't work?  I think I'm more knowledgable than the average matplotlib user, and I have no practical idea what you mean here.  I assume a list or np.array is OK?   What is a "generator expression"?  Can you give an example.

Reply:
What in particular do you find problematic? `iter(iterable) is not iterable` or "copyable"? 


Technically, we must be able to call:
~~~
for elem in iterable:
    # do something
~~~
mutliple times for `repeat=True`. While this works with many common containers like lists and numpy arrays, it's not true for every iterable. Thing is that these non-reusable iterables are all more or less advanced. E.g.

- `iterable = iter(range(10))`
- [Generator expressions](https://www.python.org/dev/peps/pep-0289/): `iterable = (x**2 for x in range(10))`
- Generators
  ```
  def my_iter(n):
      for i in range(n):
          yield i
  iterable = my_iter(3)
  ```
- I can create my own class that implements the iterator protocol but does not fulfill the above requirement.

There are two ways to try and get around the issue:
- use `iterable2 = iter(iterable)`
- use `iterable2 = copy.copy(iterable)`
We try both, but depending on the iterable both, either one, or none of these work.

I have no idea how to express this more clearly than what I've written in the docstring. If you do, you're welcome to improve the text.

That said, this is just an additional note on some technical details. It doesn't matter too much if the novice user does not understand this.

Reply:
Something that makes it clear most people don't need to read the rest of the sentence?  Something like: 

"Note some advanced and custom iterables cannot be re-used or copied, so the default ``repeat=True`` won't work; i.e. a list or numpy array will work, but the moree advanced `iterable=iter(range(10))` may not.  Its possible to work around this by passing as `frames = iter(iterable)`."

Reply:
> Its possible to work around this by passing as `frames = iter(iterable)`.

Unfortunately, this is not true. There are iterables, where `iter(iterable) is iterable` (e.g. `iter(range(10))`). If this was possible, I would have used it internally, and could have saved all the hassle of describing that not all iterables can be made to work with `repeat=True`.

Note also, that `iter(range(10))` is not something you would find in real-life code. Within our discussion and the original bug report #13676, it just serves as an over-simplified example for an arbitrary iterator. As such I wouldn't want to use it in the docs.

Reply:
@jklymak I've rephrased the section please check if that's more understandable.